### PR TITLE
Hotfix/16974 can see description diff without authorization

### PR DESCRIPTION
**Pull request by floriank**
This should fix a problem where any user could view the description diff of a Work Package.

This was reported in https://community.openproject.org/work_packages/16974

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://floriank-hotfix-16974-can-see-description-diff-without-authoriz.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-hotfix-16974-can-see-description-diff-without-authoriz.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: spec/controllers/journals_controller_spec.rb [hunk 1]
```diff
@@ -28,49 +28,70 @@
...
     end
 
-    it { expect(response).to be_success }
-    it { expect(response.body.strip).to eq("<div class=\"text-diff\">\n  <ins class=\"diffmod\">description</ins>\n</div>") }
+    describe 'w/ authorization' do
+      it 'should be successful' do
+        expect(response).to be_success
+      end
+
+      it 'should presetn the diff correctly' do
```
Comment (myabc) [inline spec/controllers/journals_controller_spec.rb:72]: @floriank typo

Comment (myabc) [inline spec/controllers/journals_controller_spec.rb:72]: (not yours though :wink:)

**Comment by ulferts** [comment]
@floriank this fixes the issue but raises a new one. The :diff action on the journals controller is currently only accessible for users having the :view_work_packages permission.

The diff method on the other hand aims to be generic, in the sense of it not being limited to returning the diff of work package description (although the visible_field? method already limits this). With the permission being what it is, this now fully becomes limited to the work package diff use case.

I am willing to live with the limitation if we make it explicit. Therefore I propose to return a 404 if the journable/journable_type isn't a work package.

**Pull request merged**